### ISSUE TITLE
Transition started and finished callbacks

### ISSUE THREAD

**Issue by guvenakcoban** [tailwindlabs/headlessui#16]
I use the Transition component for my custom modal and I need to clear the component state when the modal completely is hidden.

onEnterStarted, onEnterFinished, onLeaveStarted, onLeaveFinished methods could be a nice feature!

**Comment by ThiagoFelippi** [comment]
I could implement this?

**Comment by RobinMalfait** [comment]
Hey! Thank you for your feature suggestion!
Much appreciated! 🙏

We will expose some events, but probably not all. I'll link to this issue when I open a PR!

**Comment by RobinMalfait** [comment]
Hey @guvenakcoban I've opened a PR here: #57 what do you think?

**Comment by guvenakcoban** [comment]
@RobinMalfait Looks good! Thanks for your hard work.

**Comment by tgroshon** [comment]
It looks like this got merged to `devellop` but never released?  I don't see any of the relevant code in the actual source published on NPM.

**Comment by RobinMalfait** [comment]
Correct, didn't write the docs for it yet. You can test it out using the dev build `@headlessui/react@dev`
